### ISSUE TITLE
Indent Rule - Possible to add LogicalExpression for multi-line conditions?

### ISSUE THREAD

**Issue by ashclarke** [eslint/eslint#8978]
**What rule do you want to change?**

Add `LogicalExpression` as a configuration option to the Indent rule.

I believe the default should be "off", as per [this comment](https://github.com/eslint/eslint/blob/master/lib/rules/indent.js#L1005).

**Does this change cause the rule to produce more or fewer warnings?**

Code and setting dependent. I think the addition would not have an affect to default behaviour, unless explicitly set in eslintrc.json.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New `LogicalExpression` option for the [Indent](eslint.org/docs/rules/indent) rule configuration

**Please provide some example code that this change will affect:**

```
"indent": [2, 2, {
    "LogicalExpression": "first"
}
```

```js
// Valid
if (this.condition) {

}

// Valid
if (this.condition &&
    this.otherConditionFunctionName() &&
    this.anotherConditionFunctionName()) {

}

// Valid
var isAConditionalVariable = this.condition &&
                             this.otherConditionFunctionName();

// Invalid
if (this.condition &&
  this.otherConditionFunctionName() &&
  this.anotherConditionFunctionName()) {

}

// Invalid
var isAConditionalVariable = this.condition &&
  this.otherConditionFunctionName();
```

```
"indent": [2, 2, {
    "LogicalExpression": 1
}
```

```js
// Valid
if (this.condition) {

}

// Valid
if (this.condition &&
  this.otherConditionFunctionName() &&
  this.anotherConditionFunctionName()) {

}

// Valid
var isAConditionalVariable = this.condition &&
  this.otherConditionFunctionName();

// Invalid
if (this.condition &&
    this.otherConditionFunctionName() &&
    this.anotherConditionFunctionName()) {

}

// Invalid
var isAConditionalVariable = this.condition &&
                             this.otherConditionFunctionName();
```

*Default*

```
"indent": [2, 2, {
    "LogicalExpression ": "off"
}
```

```js
// Valid
if (this.condition) {

}

// Valid
if (this.condition &&
      this.otherConditionFunctionName() &&
        this.anotherConditionFunctionName()) {

}

// Valid
var isAConditionalVariable = this.condition &&
                                  this.otherConditionFunctionName();
```

**What does the rule currently do for this code?**

It does not check these expressions, and adding `LogicalExpression` to the rule config object raises errors about additional properties.

**What will the rule do after it's changed?**

Check additional AST nodes for indentation.

Edit: Filled out the rule change template - Thanks for linking that; apologies for not seeing it.

**Comment by not-an-aardvark** [comment]
Thanks for the proposal, but I'm confused. Isn't your example already valid with the `indent` rule?

Also, I think the term you're looking for is `BinaryExpression`. `ConditionalExpression` is something else (it refers to expressions of the form `a ? b : c`).

**Comment by ashclarke** [comment]
Ah, forgive me. I'm still an AST node newbie.

`BinaryExpression` isn't in the indent rule either, however.

**Comment by ashclarke** [comment]
Just noticed your mention about the template. I've updated the issue. Thanks!

(And sorry for not noticing it was available!)

**Comment by not-an-aardvark** [comment]
Sorry about the delayed response here. This is something I've been working on separately from this issue -- I think it's a good idea, but we will need to make sure we can handle all of the cases in a reasonable way.

**Comment by ashclarke** [comment]
No worries! I look forward to any news 👍🏻

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.
